src/varlinkctl/varlinkctl.c

```diff
@@ -289,56 +292,129 @@ typedef struct GetInterfaceDescriptionData {
+                arg_json_format_flags |= JSON_FORMAT_SEQ;
+
+        _cleanup_strv_free_ char **methods = NULL;
+
+        STRV_FOREACH(i, interfaces) {
+                JsonVariant *reply = NULL;
+                r = varlink_callb_and_log(
+                                vl,
+                                "org.varlink.service.GetInterfaceDescription",
+                                &reply,
+                                JSON_BUILD_OBJECT(JSON_BUILD_PAIR_STRING("interface", *i)));
                 if (r < 0)
                         return r;
 
-                /* Try to parse the returned description, so that we can add syntax highlighting */
-                r = varlink_idl_parse(ASSERT_PTR(description), &line, &column, &vi);
-                if (r < 0) {
-                        log_warning_errno(r, "Failed to parse returned interface description at %u:%u, showing raw interface description: %m", line, column);
+                if (FLAGS_SET(arg_json_format_flags, JSON_FORMAT_OFF) || list_methods) {
+                        static const struct JsonDispatch dispatch_table[] = {
+                                { "description",  JSON_VARIANT_STRING, json_dispatch_const_string, 0, JSON_MANDATORY },
+                                {}
+                        };
+                        _cleanup_(varlink_interface_freep) VarlinkInterface *vi = NULL;
+                        const char *description = NULL;
+                        unsigned line = 0, column = 0;
 
-                        fputs(description, stdout);
-                        if (!endswith(description, "\n"))
-                                fputs("\n", stdout);
+                        r = json_dispatch(reply, dispatch_table, JSON_LOG|JSON_ALLOW_EXTENSIONS, &description);
+                        if (r < 0)
+                                return r;
+
+                        if (!list_methods && i > interfaces)
+                                print_separator();
+
+                        /* Try to parse the returned description, so that we can add syntax highlighting */
+                        r = varlink_idl_parse(ASSERT_PTR(description), &line, &column, &vi);
+                        if (r < 0) {
+                                if (list_methods)
+                                        return log_error_errno(r, "Failed to parse returned interface description at %u:%u: %m", line, column);
+
+                                log_warning_errno(r, "Failed to parse returned interface description at %u:%u, showing raw interface description: %m", line, column);
+
+                                pager_open(arg_pager_flags);
+                                fputs(description, stdout);
+                                if (!endswith(description, "\n"))
+                                        fputs("\n", stdout);
+
+                        } else if (list_methods) {
+                                for (const VarlinkSymbol *const*symbol = vi->symbols; *symbol; symbol++) {
```

Comment:
```suggestion
                                for (const VarlinkSymbol * const *i = vi->symbols, *symbol; (symbol = *i); symbol++) {
```

and then there's no need to dereference every time `symbol` is used?